src/codegen/c.zig

```diff
@@ -158,18 +198,18 @@ fn genFn(file: *C, decl: *Decl) !void {
         try writer.writeAll("\n");
         for (instructions) |inst| {
             if (switch (inst.tag) {
-                .assembly => try genAsm(&ctx, inst.castTag(.assembly).?),
-                .call => try genCall(&ctx, inst.castTag(.call).?),
-                .add => try genBinOp(&ctx, inst.cast(Inst.BinOp).?, "+"),
-                .sub => try genBinOp(&ctx, inst.cast(Inst.BinOp).?, "-"),
+                .assembly => try genAsm(&ctx, file, inst.castTag(.assembly).?),
```

Comment:
Why remove `file` from the context if you're just going to pass it to nearly everything anyways?

Reply:
The `file` contains everything needed to output C source when the CBE is selected (buffers for main, constants, etc). That doesn't apply when we're generating a header and we may not even have one since we could be using a different backend in combination with `emit-h`.

Therefore, I've removed it from the context and passed it as a separate argument in the non-header code paths. Another idea could be to make it optional within the context but that seemed more error-prone to me.